pydantic/tools.py

```diff
@@ -0,0 +1,48 @@
+from functools import lru_cache
+from pathlib import Path
+from typing import Any, Callable, Type, TypeVar, Union
+
+from pydantic.parse import Protocol, load_file
+
+from .typing import display_as_type
+
+__all__ = ('parse_obj',)
+
+NameGenerator = Callable[[Type[Any]], str]
```

Comment:
```suggestion
NameFactory = Callable[[Type[Any]], str]
```
Would make more sense.

Reply:
but it's only ever used with a union on str, maybe better to have

```py
NameType = Union[None, str, Callable[[Type[Any]], str]]
```
?